main.cc

```diff
@@ -152,6 +144,31 @@ int main(){
 	return 0;
 }
 
+std::string getToken() {
+	std::string token;
+	
+		/*
+			First attempt to read the token from the BOT_TOKEN environment variable.
+		*/
+	char const* env = std::getenv("BOT_TOKEN");
+	if(env != nullptr) {
+		token = (std::string) env;
+	} else {
+		/*/
+		 * Read token from token file.
+		 * Tokens are required to communicate with Discord, and hardcoding tokens is a bad idea.
+		 * If your bot is open source, make sure it's ignore by git in your .gitignore file.
+		/*/
+		std::ifstream tokenFile("token.dat");
+		if(!tokenFile){
```

Comment:
I'm trying to remember: do you need to close a file even if it fails?

Reply:
In this case it doesn't matter because ifstream handles it internally in the destructor